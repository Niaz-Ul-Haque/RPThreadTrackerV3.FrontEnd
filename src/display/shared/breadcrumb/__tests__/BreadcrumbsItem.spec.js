// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import BreadcrumbsItem from '../BreadcrumbsItem';
// #endregion imports

jest.mock('../_routes', () => ({
	'/test': 'Test Page',
	'/test/test2': 'Test 2 Page'
}));

const createTestProps = propOverrides => ({
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot if route does not exist', () => {
			const props = createTestProps({ match: { url: '/test/test3' } });
			const jsx = (<BreadcrumbsItem {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot if route is exact', () => {
			const props = createTestProps({ match: { url: '/test/test2', isExact: true } });
			const jsx = (<BreadcrumbsItem {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot if route is not exact', () => {
			const props = createTestProps({ match: { url: '/test/test2', isExact: false } });
			const jsx = (<BreadcrumbsItem {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
	});
	describe('content', () => {
		it('should render null if route does not exist', () => {
			const props = createTestProps({ match: { url: '/test/test3' } });
			const jsx = (<BreadcrumbsItem {...props} />);
			const element = shallow(jsx);
			expect(element.children('BreadcrumbItem')).toHaveLength(0);
		});
		it('should render without link if route is exact', () => {
			const props = createTestProps({ match: { url: '/test/test2', isExact: true } });
			const jsx = (<BreadcrumbsItem {...props} />);
			const element = shallow(jsx);
			const item = element.find('BreadcrumbItem');
			expect(item).toHaveLength(1);
			expect(item).toHaveProp('children', 'Test 2 Page');
			expect(item.find('Link')).toHaveLength(0);
		});
		it('should render with link if route is not exact', () => {
			const props = createTestProps({ match: { url: '/test/test2', isExact: false } });
			const jsx = (<BreadcrumbsItem {...props} />);
			const element = shallow(jsx);
			const item = element.find('BreadcrumbItem');
			expect(item).toHaveLength(1);
			const link = item.find('Link');
			expect(link).toHaveLength(1);
			expect(link).toHaveProp('children', 'Test 2 Page');
			expect(link).toHaveProp('to', '/test/test2');
			expect(link).toHaveProp('href', '/test/test2');
		});
	});
});
