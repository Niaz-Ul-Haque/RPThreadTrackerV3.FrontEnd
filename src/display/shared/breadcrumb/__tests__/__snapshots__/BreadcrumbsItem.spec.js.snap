// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot if route does not exist 1`] = `""`;

exports[`rendering snapshots should render valid snapshot if route is exact 1`] = `
<BreadcrumbItem
  active={true}
>
  Test 2 Page
</BreadcrumbItem>
`;

exports[`rendering snapshots should render valid snapshot if route is not exact 1`] = `
<BreadcrumbItem>
  <Link
    href="/test/test2"
    replace={false}
    to="/test/test2"
  >
    Test 2 Page
  </Link>
</BreadcrumbItem>
`;
